transport/http/agent.go

```diff
@@ -0,0 +1,123 @@
+package http
+
+import (
+	"net/http"
+	"sync"
+
+	"go.uber.org/yarpc/internal/errors"
+	"go.uber.org/yarpc/transport"
+	"go.uber.org/yarpc/transport/peer/hostport"
+)
+
+// Agent keeps track of http peers and the associated client with which the peer will call into.
+type Agent struct {
+	sync.Mutex
+
+	client    *http.Client
+	peerNodes map[string]*peerNode
```

Comment:
PeerIdentifier.Identifier() it's the only common function across different peerIdentifiers and should be shared across peerIdentifiers no matter what the underlying implementation is